modules/ipi-install-modifying-install-config-for-slaac-dual-stack-network.adoc

```diff
@@ -0,0 +1,52 @@
+// This is included in the following assemblies:
+//
+// ipi-install-configuration-files.adoc
+
+:_content-type: PROCEDURE
+[id='ipi-install-modifying-install-config-for-slaac-dual-stack-network_{context}']
+= Optional: Configuring address generation modes for SLAAC in dual-stack networks
+
+For dual-stack clusters that use Stateless Address AutoConfiguration (SLAAC), you must specify a global value for the `ipv6.addr-gen-mode` network setting. You can set this value by configuring the ramdisk and the cluster configuration files. If you don't configure a consistent `ipv6.addr-gen-mode` in these locations, IPv6 address mismatches can occur between CSR resources and `BareMetalHost` resources in the cluster. 
+
+.Prerequisites
+
+* You have installed the `butane` utility.
+
+.Procedure
+
+. Create a Butane configuration file, `addr-more-nm-override.bu`, to define the `ipv6.addr-gen-mode` value:
++
+[source,yaml]
+----
+variant: openshift
+version: 4.13.0
+metadata:
+  name: 98-addr-mode-nm-override
+  labels:
+    machineconfiguration.openshift.io/role: master
+storage:
+  files:
+    - path: /etc/NetworkManager/system-connections/default.nmconnection
+      mode: 0600
+      overwrite: true
+      contents:
+        inline: |
+          # The Machine Config Operator manages this file.
+          server openshift-master-0.<cluster-name>.<domain> iburst <1>
+          server openshift-master-1.<cluster-name>.<domain> iburst
+          server openshift-master-2.<cluster-name>.<domain> iburst
+
+          [ipv6]
+          addr-gen-mode=<address_mode> <2>          
+----
+<1> You must replace `<cluster-name>` with the name of the cluster and replace <domain> with the fully qualified domain name.
+<2> You must replace `<address_mode>` with the type of address generation mode required for IPv6 addresses in the cluster. Valid values are `eui64`, `stable-privacy`, or `random`.
+
+. Use Butane to generate a `MachineConfig` object file, `addr-more-nm-override.yaml`, containing the configuration for the worker nodes:
++
+[source,terminal]
+----
+$ butane addr-more-nm-override.bu -o addr-more-nm-override.yaml
```

Comment:
I'm not sure butane needs to be used here. Doesn't that generate ignition? Machine-configs are converted to ignition by MCO.

Reply:
Hi @cybertron, should butane not be used at all in this procedure? I was replicating other [procedures](https://file.emea.redhat.com/rohennes/OCPBUGS-9194-CSR/installing/installing_bare_metal_ipi/ipi-install-installation-workflow.html#configuring-ntp-for-disconnected-clusters_ipi-install-installation-workflow) in that section of the docs, as I thought it was required for configuring networking pre-installation for IPI. Perhaps I am misunderstanding and it is only the ramdisk part that needs to be configured for Day 0?

Should I remove butane from this procedure and use a regular yaml file like below that the user applies after the cluster is running? (And move to a post-installation section of the docs) 

````
apiVersion: machineconfiguration.openshift.io/v1
kind: MachineConfig
metadata:
  name: 90-ipv6-addr-gen-mode-override
spec:
  config:
    ignition:
      version: 3.2.0
    storage:
       files:
          - path: /etc/NetworkManager/conf.d/90-ipv6-addr-gen-mode-override.conf
            mode: 0600
            contents:
               inline: |
                   [connection-90-ipv6-addr-gen-mode-override]
                   match-device=type:ethernet
                   ipv6.addr-gen-mode=<address_mode>

````

Reply:
Thanks @cybertron - I removed Butane. Perhaps I was misunderstanding how pre-installation custom configs worked. This any better?